Feat/footer

pavlo1031 opened the pull request.

pavlo1031 committed:
feat: footer layout init

[目前]:
• logo圖是使用原本專案的
• footer的曲線圖案是另外加入: @/images/footer-curve-line.svg
• css寫在外面: @/components/Footer.css (目前和Footer.tsx放同一層)

[待調整]:
• 樣式改為使用Tailwind CSS
  (Footer.css未來會消失? 移至別處?)
• 引用圖片、連結、...時，改用React寫法
• 其他微調

pavlo1031 committed:
feat: remove 'async' modifier on Footer( )

pavlo1031 committed:
feat: footer : add tailwind classes

在研究如何套用 background image
(Footer的logo、曲線圖案)

pavlo1031 committed:
refactor: 改以 tailwind classes 的方式, 呈現 footer title , curve line

• 加上spacing：14, 16, 20, 21 (用在footer上方距離)
• Logo : background image ➜ 改使用<Image>
• 超連結: <a> ➜ 改使用<Link>

[待調整]:
• 加上javascript, 使得「無法按右鍵另存圖片」？

pavlo1031 committed:
style: 修正: 把 tab ➜ 改為' ' (空格)

pavlo1031 committed:
feat: add event handler disabling right click menu

• Avoid users do 'Save Image as'
  (on the footer title image and the curve line image)

pavlo1031 committed:
chore: delete unused footer css file

pavlo1031 committed:
style: adjust spacing in/between the footer

• add spacings to tailwind.config.js

pavlo1031 committed:
style: adjust width & height, margins , text sizes... for devices of each screen sizes

pavlo1031 commented:
there's still some issues to fix

pavlo1031 committed:
fix: add explicit type name on the parameter 'e' of onContextMenu( )

• This results in build failure
  error message: "Parameter 'e' implicitly has an any type."

pavlo1031 commented:
- reopen the pull request after fixing a "build failure" issue
  (add explicit type on variable name)

ba1679 commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+'use client';
```
[Suggested] 'use client' 目前專案中都有空一行
```suggestion
'use client';

```

pavlo1031 replied:
✅ 已在 2024/6/01 19:43 的 commit : a963717 修改完畢

ba1679 commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+'use client';
+import Link from 'next/link';
+import Image from 'next/image';
+import MouseEvent from 'react';
```
[Required] MouseEvent 是從 'react' 解構出來的
```suggestion
import { MouseEvent } from 'react';
```

pavlo1031 replied:
✅ 已在 2024/6/01 19:38 的 commit : 5880082  修改完畢
      （引入 MouseEventHandler時，一併和 MouseEvent 改以此種 import方式）

ba1679 commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+export function Footer() {
+
+  const onContextMenuHandler = (e: MouseEvent) => {
+    if(e.preventDefault != undefined)
+      e.preventDefault();
+    if(e.stopPropagation != undefined)
+      e.stopPropagation();
+    return false;
```
[Suggested] 有了 TypeScript 這邊的檢查就不需要了，然後感覺只需要 e.preventDefault() ?
```suggestion
      e.preventDefault();
```

pavlo1031 replied:
1. "這邊的檢查就不需要了"
妳指的是 對於 **e.preventDefault != undefined** 這樣的檢查嗎？
2. "感覺只需要 e.preventDefault() "
➜ 已在 2024/6/01 19:43 的 commit : a963717  修改完畢

CoolizzLuo replied:
不用判斷是不是 undefined，因為已經加上 type

pavlo1031 replied:
@CoolizzLuo, @ba1679 
➜ 已在 2024/6/02 08:58 的 commit : 56da986 移除此判斷 : "**undefined check**"

ba1679 commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+
+  return (
+    <footer className='mt-18 py-13 pb-14 gap-5 bg-grey flex flex-col justify-center items-center'>
+      <div className='footer-title' onContextMenu={onContextMenuHandler}>
+        <Image src={Logo} className='object-cover object-center w-[200px] sm:w-[240px] md:w-[240px] lg:w-[245px] xl:w-[250px]' alt='Journey Bites' priority />
+      </div>
+      <div className='footer-curve-line mt-[-6px] mb-3 lg:mb-3.5' onContextMenu={onContextMenuHandler}>
+        <Image src={CurveLine} alt='' />
+      </div>
```
[Required]  
1. 沒用到的 className 要移除
2. 直接用 Image 就好，外面不用再多一層 div (改了之後上面 onContextMenuHandler 要改成 HTMLImageElement)
```suggestion
       <Image src={Logo} className='object-cover object-center w-[200px] sm:w-[240px] md:w-[240px] lg:w-[245px] xl:w-[250px]' onContextMenu={onContextMenuHandler} alt='Journey Bites' priority />
        <Image src={CurveLine} className='mt-[-6px] mb-3 lg:mb-3.5' onContextMenu={onContextMenuHandler} alt='' />
```

pavlo1031 replied:
1. "沒用到的className要移除"
➜ 已在 2024/6/02 08:03 的 commit : 211ac0b  修改完畢
2. "直接用 Image 就好，外面不用再多一層 div"
➜ 已在 2024/6/02 08:38 的 commit : 79b058f  修改完畢

CoolizzLuo replied:
alt 都要寫不要留空字串

pavlo1031 replied:
@CoolizzLuo 所以 footer curve line 它的<Image> 的 alt 也要寫 'Jouney Bites' ？

pavlo1031 replied:
「為 <Image> 的 alt屬性都加上值」這個程式碼改動
 commit message **prefix** 要寫什麼啊？

CoolizzLuo replied:
要寫出這個圖片是幹嘛，像是 logo 就寫 logo 方便 SEO 閱讀，也方便在圖片沒有顯示的時候有文字知道這是做什麼
改動的話寫 `fix` 就好了

pavlo1031 replied:
那 footer curve line 的 alt 該寫什麼呢？
那個純粹是裝飾啊

CoolizzLuo replied:
就寫個  wave 或 curve 之類都可以

pavlo1031 replied:
> https://github.com/Journey-bites/journey-bites-website/pull/17#discussion_r1623304628
> "alt 都要寫不要留空字串"

已在 2024/6/02 09:42 的 commit : c205505  修改完畢 ➜ **<img>都加上alt值了**

pavlo1031 replied:
> "alt 都要寫不要留空字串"

➜ 已在 2024/6/02 09:42 的 commit : c205505  修改完畢

ba1679 commented on src/components/Footer.tsx:
```diff
@@ -0,0 +1,36 @@
+      <div className='footer-title' onContextMenu={onContextMenuHandler}>
+        <Image src={Logo} className='object-cover object-center w-[200px] sm:w-[240px] md:w-[240px] lg:w-[245px] xl:w-[250px]' alt='Journey Bites' priority />
+      </div>
+      <div className='footer-curve-line mt-[-6px] mb-3 lg:mb-3.5' onContextMenu={onContextMenuHandler}>
```
為什麼要 mt 負數 mb 又正數？

pavlo1031 replied:
用了不太正統的方式微調
➜ 待重新調整，一處一處的和Figma校對

ba1679 commented on tailwind.config.ts:
```diff
@@ -28,6 +28,13 @@ const config = {
       12: '48px',
+      13: '42px',
+      14: '56px',	
+      16: '64px',
+      18: '72px',	
+      19: '76px',	
+      20: '80px',
+      21: '96px'
```
[Suggested] 看起來目前只有用到 13, 14，後面可以先不用加

ba1679 replied:
[Required] 1 個單位是 4px，13 的話會是 13 * 4

pavlo1031 replied:
- 👌 已排入我的待修改清單

pavlo1031 committed:
fix: lint error: (type mismatch wiith <div>'s property 'onContextMenu'

1.Remove the return statement.
2.Add explicit type declaration 'MouseEventHandler' on the function onContextMenuHandler.

pavlo1031 committed:
style: it's a code style change according to Render Lai's suggestion

• a new line after 'use client'
• calling just e.preventDefault() is enough to prevent default right-click behavior.

pavlo1031 commented:
there's still other something to fix.

pavlo1031 commented:
- fix issues (mostly eslint issues)
- apply coding suggestions from Render Lai.

pavlo1031 committed:
refactor: change code style according  to Render Lai's suggestion.

• Remove unnecessary classes
  (i.e. footer-title, footer-curve-line, ...)

pavlo1031 committed:
refactor: adjust element structure.

• remove the external wrapping <div> element of <Image>

pavlo1031 committed:
refactor: remove unnecessary "undefined check"

• "undefined check" is unnecessary because we've added its type 'MouseEvent'

pavlo1031 committed:
fix: add alt attribute values for all the <Image> tags

ba1679 commented:
Due to a reallocation of tasks, I will be taking over the implementation of the footer UI in another PR.
